docs: restructure working PR (#4866)
dfinity/motoko: doc/md/fundamentals/1-basic-syntax/9-operators.md

Bitwise operators manipulate numbers **at the binary level**.

| Operator | Description |Example |
|----------|------------|----------|
| `&`      | Bitwise AND |`a & b` |
|  `|`    | Bitwise OR | `a | b` |
| `^`      | Bitwise XOR | `a ^ b` |
| `<<`     | Shift left | `a << b` |
| `>>`     | Shift right (must be proceeded by whitespace) |`a >> b` |
| `<<>`    | Rotate left (circular shift) | `a <<> b` |
| `<>`     | Rotate right (circular shift)| `a <> b` |

:::caution

Bitwise operators can only be used with bounded types.

:::

## Wrapping operators

Bounded integers **trap** on overflow, but **wrapping versions** allow overflow behavior.

| Operator | Description | Example |
|----------|------------|------------|
| `+%`     | Addition with wrap-around | `a +% b` |
| `-%`     | Subtraction with wrap-around | `a -% b` |
| `*%`     | Multiplication with wrap-around | `a *% b` |
| `**%`    | Exponentiation with wrap-around | `a **% b` |

## Text operators

| Operator | Description | Example |
|----------|------------|------------|
| `#`      | Concatenates two [`Text`](https://internetcomputer.org/docs/motoko/base/Text) values | `a # b` |

## Assignment operators

> Assignment operators modify variables in place. Only mutable variables `var` can be reassigned.

Reviewer: ```suggestion
Assignment operators modify variables in place. Both mutable variables declared with `var` and elements of mutable arrays can be assigned new values.
```

| Operator | Description |Examples|
|----------|------------|---------|
| `:=`     | Assign a value | `a := b` |
| `+=`     | Add and assign | `a += b` |
| `-=`     | Subtract and assign | `a -= b` |
| `*=`     | Multiply and assign | `a *= b` |
| `/=`     | Divide and assign | `a /= b` |
| `#=`     | Concatenate and assign (for [`Text`](https://internetcomputer.org/docs/motoko/base/Text)) | `a #= b` |

> 

Reviewer: ````suggestion
For example:

```motoko
var done = false; 
done := true;

let a = [var 1, 2];
a[0] += a[1];
```
````

## Operator precedence

> Operators follow precedence rules, meaning some operators are evaluated before others.

Reviewer: ```suggestion
Operators follow precedence rules, meaning that, in the absence of explicit parentheses, some operators are evaluated before others.
```

1. Unary operators (`-`, `!`, `^`)
2. Exponentiation (`**`, `**%`)
3. Multiplication & division (`*`, `/`, `%`, `*%`)
4. Addition & subtraction (`+`, `-`, `+%`, `-%`)
5. Bitwise operators (`&`, `|`, `^`)
6. Comparison operators (`==`, `!=`, `<`, `>`, `<=`, `>=`)
7. Assignment operators (`:=`, `+=`, `-=`, etc.)

For example:

```motoko
let result = 10 + 5 * 2; // result = 20
```

Use parentheses to enforce a different order.

```motoko
let result = (10 + 5) * 2; // result = 30
```

> ## Pipes

Reviewer 2: Would a section on `do ? { ... option! ... }` fit here with other operators? This is a like an unary operator
